lib/core/include/apiHandler.hpp

```diff
@@ -143,56 +142,70 @@ namespace irods {
                                 re_plugin_globals->global_re_mgr,
                                 &rei );
 
-                        for ( auto& ns : NamespacesHelper::Instance()->getNamespaces() ) {
-                            std::string rule_name = ns + "pep_" + operation_name + "_pre";
-                            if ( RuleExistsHelper::Instance()->checkOperation( rule_name ) ) {
-                                if ( re_ctx_mgr.rule_exists( rule_name, ret ).ok() && ret ) {
-                                    op_err = re_ctx_mgr.exec_rule( rule_name, "api_instance", ctx, std::forward<types_t>(_t)... );
-                                    if ( !op_err.ok() ) {
-                                        rodsLog( LOG_DEBUG, "Pre-pep rule [%s] failed with error code [%d]", rule_name.c_str(), op_err.code() );
-                                        saved_op_err = op_err;
-                                        pre_pep_failed = true;
-                                    }
-                                } else {
-                                    rodsLog( LOG_DEBUG, "Rule [%s] passes regex test, but does not exist", rule_name.c_str() );
-                                }
+                        // invoke the pre-pep for this operation
+                        error pre_err = invoke_policy_enforcement_point(
+                                            re_ctx_mgr,
+                                            ctx,
+                                            operation_name,
+                                            "pre",
+                                            std::forward<types_t>(_t)...);
+                        if(!pre_err.ok()) {
+                            // if the pre-pep fails, invoke the exception pep
+                            error except_err = invoke_policy_enforcement_point(
+                                               re_ctx_mgr,
+                                               ctx,
+                                               operation_name,
+                                               "except",
+                                               std::forward<types_t>(_t)...);
+                            if(!except_err.ok()) {
+                                irods::log(PASS(except_err));
```

Comment:
Did we want to return except_err in this case, or is it more important that the caller knows which PEP initially failed and why? Same for 179 and 200, and in the plugin_base.

Also, would it be worth capturing "pre", "post", and "except" in consts?